crates/bevy_core_pipeline/src/contrast_adaptive_sharpening/mod.rs

```diff
@@ -0,0 +1,265 @@
+impl Default for ContrastAdaptiveSharpeningSettings {
+    fn default() -> Self {
+        ContrastAdaptiveSharpeningSettings {
+            enabled: true,
+            contrast_adaption: 0.1,
+            sharpening_intensity: 1.0,
+        }
+    }
+}
+
+impl ExtractComponent for ContrastAdaptiveSharpeningSettings {
+    type Query = &'static Self;
+    type Filter = With<Camera>;
+    type Out = CASUniform;
+
+    fn extract_component(item: QueryItem<Self::Query>) -> Option<Self::Out> {
+        if !item.enabled {
+            return None;
+        }
+        Some(CASUniform {
+            contrast_adaption: item.contrast_adaption.clamp(0.0, 1.0),
+            sharpening_intensity: item.sharpening_intensity.clamp(0.0, 1.0),
+        })
+    }
+}
+
+const CONTRAST_ADAPTIVE_SHARPENING_SHADER_HANDLE: HandleUntyped =
+    HandleUntyped::weak_from_u64(Shader::TYPE_UUID, 6925381244141981602);
+
+/// Adds Support for Contrast Adaptive Sharpening (CAS).
+pub struct ContrastAdaptiveSharpeningPlugin;
+
+impl Plugin for ContrastAdaptiveSharpeningPlugin {
+    fn build(&self, app: &mut App) {
+        load_internal_asset!(
+            app,
+            CONTRAST_ADAPTIVE_SHARPENING_SHADER_HANDLE,
+            "contrast_adaptive_sharpening.wgsl",
+            Shader::from_wgsl
+        );
+
+        app.register_type::<ContrastAdaptiveSharpeningSettings>();
+        app.add_plugin(ExtractComponentPlugin::<ContrastAdaptiveSharpeningSettings>::default());
+        app.add_plugin(UniformComponentPlugin::<CASUniform>::default());
+
+        let render_app = match app.get_sub_app_mut(RenderApp) {
+            Ok(render_app) => render_app,
+            Err(_) => return,
+        };
+        render_app
+            .init_resource::<ContrastAdaptiveSharpeningPipeline>()
+            .init_resource::<SpecializedRenderPipelines<ContrastAdaptiveSharpeningPipeline>>()
+            .add_system_to_stage(
+                RenderStage::Prepare,
+                prepare_contrast_adaptive_sharpening_pipelines,
+            );
+        {
+            let cas_node = ContrastAdaptiveSharpeningNode::new(&mut render_app.world);
+            let mut binding = render_app.world.resource_mut::<RenderGraph>();
+            let graph = binding.get_sub_graph_mut(core_3d::graph::NAME).unwrap();
+
+            graph.add_node(core_3d::graph::node::CONTRAST_ADAPTIVE_SHARPENING, cas_node);
+
+            graph.add_slot_edge(
+                graph.input_node().id,
+                core_3d::graph::input::VIEW_ENTITY,
+                core_3d::graph::node::CONTRAST_ADAPTIVE_SHARPENING,
+                ContrastAdaptiveSharpeningNode::IN_VIEW,
+            );
+
+            graph.add_node_edge(
+                core_3d::graph::node::FXAA,
+                core_3d::graph::node::CONTRAST_ADAPTIVE_SHARPENING,
+            );
+            graph.add_node_edge(
+                core_3d::graph::node::CONTRAST_ADAPTIVE_SHARPENING,
+                core_3d::graph::node::END_MAIN_PASS_POST_PROCESSING,
```

Comment:
This should maybe be tonemapping? In general, the way we order post processing probably needs improving.

Reply:
Tonemapping comes before FXAA, and sharpening should happen after FXAA, so that wouldn't work.

Reply:
Hmm, I apply TAA before tonemapping and bloom and some other stuff. Should CAS go immediately after TAA, or before tonemapping?

Reply:
Sharpening should probably be one of the last post processing passes, to avoid any sharpening artifacts screwing up the output of a different pass. (Ideally it would happen after upscaling, if/when we get actual upscaling).